enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java

```diff
@@ -66,4 +78,80 @@ public static EndpointCluster processEndpoints(org.wso2.choreo.connect.discovery
+        Map<String, String> headersMap = requestContext.getHeaders();
+        if (headersMap.containsKey(APIConstants.ACCEPT_HEADER.toLowerCase())) {
+            acceptType = headersMap.get(APIConstants.ACCEPT_HEADER.toLowerCase());
+        }
+
+        if (mockedApiConfig.getIn().equalsIgnoreCase(APIConstants.MockApiConstants.HEADER)) {
+            setMockApiResponse(responseObject, headersMap, mockedApiConfig, acceptType);
+        } else {
+            Map<String, String> queryParamMap = requestContext.getQueryParameters();
+            setMockApiResponse(responseObject, queryParamMap, mockedApiConfig, acceptType);
+        }
+    }
+
+    /**
+     * Prepares mock API response considering the properties map and mockedApiConfig.
+     *
+     * @param responseObject  Response object (represents response for the API call)
+     * @param propertiesMap   A map which includes values specified in headers or query parameters
+     * @param mockedApiConfig Holds the JSON values specified in the mock API implementation
+     * @param acceptType      Denotes accepted contend type as the response
+     */
+    private static void setMockApiResponse(ResponseObject responseObject, Map<String, String> propertiesMap,
+                                           MockedApiConfig mockedApiConfig, String acceptType) {
+        String requestValuePosition = mockedApiConfig.getIn();
+        String name;
+
+        // condition handles case-sensitiveness of query parameters.
+        if (requestValuePosition.equalsIgnoreCase(APIConstants.MockApiConstants.HEADER)) {
+            name = mockedApiConfig.getName().toLowerCase();
+        } else {
+            name = mockedApiConfig.getName();
+        }
+
+        if (propertiesMap == null || !propertiesMap.containsKey(name)) {
+            log.error("Response determining value not available in the mock API request.");
+            return;
+        }
+        String nameInRequest = propertiesMap.get(name);
+        List<MockedResponseConfig> responseConfigList = mockedApiConfig.getResponses();
+        // iterates over the mock API responses.
+        for (MockedResponseConfig responseConfig : responseConfigList) {
+            if (!responseConfig.getValue().equalsIgnoreCase(nameInRequest)) {
+                log.error("Mocked response config JSON does not contain the value provided in the request.");
+                return;
+            }
+            responseObject.setStatusCode(responseConfig.getCode());
+            Map<String, String> headerMap = new HashMap<>();
+            // iterates over the headers list in the mock API JSON.
+            for (MockedHeaderConfig header : responseConfig.getHeaders()) {
+                headerMap.put(header.getName(), header.getValue());
+            }
+            MockedPayloadConfig payload = responseConfig.getPayload();
+            // checks and assigns the accepted content type
+            if (acceptType.equalsIgnoreCase(APIConstants.APPLICATION_XML) &&
```

Comment:
don't we support other content types like Content-Type: text/html

Reply:
ATM we support only for above two types.